src/emulation_core/mips/instruction.rs

```diff
@@ -0,0 +1,55 @@
+pub mod instruction_types {
+
+    #[derive(Debug, Default, Clone, Copy)]
+    pub struct RType {
+        pub op: u8,
+        pub rs: u8,
+        pub rt: u8,
+        pub rd: u8,
+        pub shamt: u8,
+        pub funct: u8,
+    }
+
+    #[derive(Debug, Default, Clone, Copy)]
+    pub struct IType {
+        pub op: u8,
+        pub rs: u8,
+        pub rt: u8,
+        pub immediate: u16,
+    }
+
+    #[derive(Debug, Default, Clone, Copy)]
+    pub struct JType {
+        pub op: u8,
+        pub addr: u32,
+    }
+
+    #[derive(Debug, Default, Clone, Copy)]
+    pub struct PlaceholderType {
```

Comment:
What exactly is the placeholder type? Maybe we can define more specifically what instructions this would be used for and make this a type of its own.